### ISSUE TITLE
Specify dependency version with cargo install 

### ISSUE THREAD

**Issue by jethrogb** [rust-lang/cargo#3266]
It would be useful to be able to specify specific dependency versions (like in cargo update --precise) when doing cargo install. Since the release of openssl-sys 0.9.0, any package that depends on git2 ^0.4 no longer builds out of the box due to two versions of openssl-sys being pulled in. This prevents using cargo install on various crates such as [cargo-clone](https://github.com/JanLikar/cargo-clone/issues/9) and [cargo-tree](https://github.com/sfackler/cargo-tree/issues/11).

**Comment by alexcrichton** [comment]
I think the only real solution here is to have Cargo use lock files when working with `cargo install` (I forget if it already does). Other than that there's basically no real ergonomic solution to this that I think we'd want long-term.

**Comment by Dushistov** [comment]
@alexcrichton 

As for me, I want this feature to install tools via cargo. Like racer, clippy, rustfmt.
Because of bugs in theese tools and their great influence on development,
I need some specific versions of them, and often it is not last stable versions,
but stable -1 or -2.

It is not ergonomic to do 'git clone' of theirs repos, switch to right tag and call `cargo install` inside tree, it would be great if it is possible to specify the same line as inside
of `Cargo.toml` in `dependicies` section. Like `cargo install --spec='{git = "https://github.com/rust-lang-nursery/rustfmt", rev = "xxxx"}'`

**Comment by alexcrichton** [comment]
@Dushistov you can already select the top-level version via `cargo install --vers ...`, this issue is about selecting the versions of transitive dependencies.

**Comment by carols10cents** [comment]
Given this comment:

> I think the only real solution here is to have Cargo use lock files when working with cargo install (I forget if it already does). Other than that there's basically no real ergonomic solution to this that I think we'd want long-term.

I'm going to close this as a duplicate of https://github.com/rust-lang/cargo/issues/2263.
